src/protocol/basics/reveal.rs

```diff
@@ -166,22 +169,28 @@ mod tests {
         let world = TestWorld::new().await;
         let sh_ctx = world.contexts::<Fp31>();
         let v = sh_ctx.map(MaliciousValidator::new);
+        let m_ctx: [_; 3] = v
+            .iter()
+            .map(|v| v.context().set_total_records(10).set_total_upgrades(10))
+            .collect::<Vec<_>>()
+            .try_into()
+            .unwrap();
 
         for i in 0..10_u32 {
             let record_id = RecordId::from(i);
             let input: Fp31 = rng.gen();
 
             let m_shares = join3v(
-                zip(v.iter(), input.share_with(&mut rng))
-                    .map(|(v, share)| async { v.context().upgrade(record_id, share).await }),
+                zip(m_ctx.iter(), input.share_with(&mut rng))
+                    .map(|(m_ctx, share)| async { m_ctx.upgrade(record_id, share).await }),
             )
             .await;
 
-            let results =
-                join3v(zip(v.iter(), m_shares).map(|(v, m_share)| async move {
-                    v.context().reveal(record_id, &m_share).await
-                }))
-                .await;
+            let results = join3v(
+                zip(m_ctx.clone().into_iter(), m_shares)
```

Comment:
Do you need the into_iter() here?  `zip` takes `impl IntoIterator`.

```suggestion
                zip(m_ctx.clone(), m_shares)
```